actions.go

```diff
@@ -0,0 +1,123 @@
+package main
+
+import (
+	"errors"
+	"github.com/Terag/kadok/security"
+	"github.com/bwmarrin/discordgo"
+	"strings"
+)
+
+// ExecuteAction is the function type to implement with an Action
+type ExecuteAction func(s *discordgo.Session, m *discordgo.MessageCreate) error
+
+// ActionResolver is the interface used by Kadok to resolve a call into an action
+type ActionResolver interface {
+	GetPermission() security.Permission
+	ResolveAction(call []string) (ActionResolver, ExecuteAction)
+}
+
+// Action structure containing information and functions regarding an action
+type Action struct {
+	// The permission required to use this action. Use the EmptyPermission if no permission is required
+	Permission security.Permission
+	// The sub actions in the hierarchy. They do not inherit the permission requirement
+	SubActions map[string]ActionResolver
```

Comment:
why are `SubActions` resolvers ?

is it an action concern to 'resolves' it's own children ? shouldn't it be the resolver problem to dig into nested actions ?

Reply:
Name changed to `ActionNode` to better reflect the usages of the interface